pkg/controllers/state/cluster.go

```diff
@@ -345,7 +345,12 @@ func (c *Cluster) updateNodeUsageFromPodDeletion(podKey types.NamespacedName) {
 
 // updatePod is called every time the pod is reconciled
 func (c *Cluster) updatePod(ctx context.Context, pod *v1.Pod) error {
-	err := c.updateNodeUsageFromPod(ctx, pod)
+	var err error
+	if podutils.IsTerminal(pod) {
+		c.updateNodeUsageFromPodCompletion(client.ObjectKeyFromObject(pod))
+	} else {
+		err = c.updateNodeUsageFromPod(ctx, pod)
```

Comment:
Can we add a test for this? There are some existing tests in `state/suite_test.go` for "Node Resource Level" that I Think this would fit in.